x-pack/auditbeat/module/system/socket/_meta/docs.asciidoc

```diff
@@ -4,3 +4,127 @@ beta[]
+
+[float]
+=== Requirements
+
+Features used by the `socket` dataset require a minimum Linux kernel version
+of 3.12 (vanilla). However, some distributions have backported those features
+to older kernels. The following (non-exhaustive) lists the different
+distributions under which the dataset is known to work:
+
+[options="header"]
+|==============================================
+| Distribution   | kernel version      | Works?
+| CentOS 6.5     | 2.6.32-431.el6      | NO^<<anchor-1,[1]>>^
+| CentOS 6.9     | 2.6.32-696.30.1.el6 | &#10003;
+| CentOS 7.6     | 3.10.0-957.1.3.el7  | &#10003;
+| Debian 8       | 3.16.0-6            | &#10003;
+| Debian 9       | 4.9.0-8             | &#10003;
+| Debian 10      | 4.19.0-5            | &#10003;
+| SLES 12        | 4.4.73-5            | &#10003;
+| Ubuntu 12.04   | 3.2.0-126           | NO^<<anchor-1,[1]>>^
+| Ubuntu 14.04.6 | 3.13.0-170          | &#10003;
+| Ubuntu 16.04.3 | 4.4.0-97            | &#10003;
+| AWS Linux 2    | 4.14.138-114.102    | &#10003;
+|==============================================
+
+[[anchor-1]]
+<<anchor-1,[1]>>: These systems lack
+https://lore.kernel.org/patchwork/patch/399251/[PERF_EVENT_IOC_ID ioctl.]
+Support might be added in a future release.
+
+The dataset needs CAP_SYS_ADMIN and CAP_NET_ADMIN in order to work.
+
+[float]
+==== Kernel configuration
+
+A kernel built with the following configuration options enabled is required.
+
+- `CONFIG_KPROBE_EVENTS`: Enables the KProbes subsystem.
+- `CONFIG_DEBUG_FS`: For kernels laking `tracefs` support (<4.1).
+- `CONFIG_IPV6`.
+
+These settings are enabled by default in most distributions.
+
+The following configuration settings can prevent the dataset from starting:
+
+- `/sys/kernel/debug/kprobes/enabled` must be 1.
+- `/proc/sys/net/ipv6/conf/lo/disable_ipv6` must be 0
+(IPv6 enabled in the loopback device).
+
+[float]
+==== Running on docker
+
+The dataset can monitor the Docker host when running inside a container. However
+it needs to run on a `privileged` container with `CAP_NET_ADMIN` and IPv6
+support.
+
+[float]
+=== Configuration
+
+The following options are available for the `socket` dataset:
+
+- `socket.tracefs_path` (default: none)
+
+Must point to the mount-point of `tracefs` or the `tracing` directory inside
+`debugfs`. If this option is not specified, {beatname_uc} will look for
+the default locations: `/sys/kernel/tracing` and `/sys/kernel/debug/tracing`.
+If not found, it will attempt to mount `tracefs` and `debugfs` at their
+default locations.
+
+- `socket.flow_inactive_timeout` (default: 30s)
+
+Determines how long a flow has to be inactive to be considered closed.
+
+- `socket.flow_termination_timeout` (default: 5s)
+
+Determines how long to wait after a socket has been closed for out of order
+packets. With TCP, some packets can be received shortly after a socket is
+closed. If set too low, additional flows will be generated for those packets.
+
+- `socket.perf_queue_size` (default: 4096)
```

Comment:
I'm not sure if this option and the ones below should be prefixed with a disclaimer or not documented at all.